### ISSUE TITLE
💅 `noVueDuplicateKeys` false positive

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#9068]
### Environment information

<details>

```bash

CLI:
  Version:                      0.0.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Error:                        The configuration schema version does not match the CLI version 0.0.0
  Status:                       Loaded with errors
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               false
  VCS enabled:                  true
  HTML full support enabled:    true

Workspace:
  Open Documents:               0

```

</details>

### Rule name

lint/correctness/noVueDuplicateKeys

### Playground link

https://biomejs.dev/playground/?lintRules=noVueDuplicateKeys&code=PABzAGMAcgBpAHAAdAAgAGwAYQBuAGcAPQAiAHQAcwAiACAAcwBlAHQAdQBwAD4ACgBjAG8AbgBzAHQAIABwAHIAbwBwAHMAIAA9ACAAZABlAGYAaQBuAGUAUAByAG8AcABzADwAewAgAHQAbwBhAHMAdAA6ACAAVABvAGEAcwB0ADsAIABuAHUAbQBiAGUAcgA%2FADoAIABuAHUAbQBiAGUAcgAgAH0APgAoACkAOwAKAAoAYwBvAG4AcwB0ACAAewAgAHQAbwBhAHMAdAAgAH0AIAA9ACAAdABvAFIAZQBmAHMAKABwAHIAbwBwAHMAKQA7AAoAPAAvAHMAYwByAGkAcAB0AD4A&language=vue

### Expected result

The prop `toast` is only defined once.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
